packages/fixed-decimal/src/FixedDecimal.ts

```diff
@@ -0,0 +1,471 @@
+import type BN from 'bn.js'
+
+/**
+ * Parses a string into the constructor args for a {@link FixedDecimal}.
+ *
+ * Doesn't do any validation of the string - if it's malformed the `BigInt`
+ * construction will fail.
+ * @param value The value represented as a fixed decimal string.
+ * @param decimalPlaces The number of decimal places the result should have.
+ * @returns
+ */
+const parseFixedDecimalString = (
+  value: string,
+  decimalPlaces?: number
+): FixedDecimalCtorArgs => {
+  let [whole, decimal] = value.split('.')
+  decimal = decimal ?? ''
+  if (decimalPlaces !== undefined) {
+    decimal = decimal.padEnd(decimalPlaces, '0').substring(0, decimalPlaces)
+  }
+  return {
+    value: BigInt(`${whole}${decimal}`),
+    decimalPlaces: decimalPlaces ?? decimal.length
+  }
+}
+
+type FixedDecimalCtorArgs = {
+  value: bigint
+  decimalPlaces: number
+}
+
+/**
+ * A custom options type for the custom toLocaleString() implementation of
+ * {@link FixedDecimal} that only allows the supported subset of the
+ * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat Intl.NumberFormat options}.
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat MDN documentation for Intl.NumberFormat}
+ */
+type FixedDecimalFormatOptions = {
```

Comment:
I appreciate the thoroughness here of supporting a localeString interface. I'm just wondering what motivated all that complexity? I don't think we use most of that functionality anywhere...

Reply:
I only implemented the parts of `toLocaleString()` that we use - min/max fractional digits, comma groupings, and rounding methods. The extra complexity of supporting other locales was an easy bonus win.

The main motivations:
- A familiar dev-ex where newcomers don't have to learn our vocabulary and nomenclature
- Existing standard of docs and usage makes the architecture/design of the function fairly battle tested, and makes it unlikely to have a need for a new format helper down the line
- I really liked the interface, especially the `roundingMode` and `trailingZeroDisplay` options

The things I don't like are having to set the locale every time and the length of the name, and there's no shorthand to do both min and max at the same time.